flask_fillin/wrapper.py

```diff
@@ -54,6 +54,12 @@ def _click(self, client, **kwargs):
             def _submit(self, client, path=None, **kargs):
                 data = dict(self.form_values())
 
+                # note that most browsers submit '' instead of None for empty fields
+                blank_fields = ['text', 'password']
+                for k, v in self.fields.iteritems():
+                  if k not in data and self.inputs[k].type in blank_fields:
+                    data[k] = '' 
+
```

Comment:
@tianhuil the fix is here - `self.form_values()` only returns nonempty fields in the first place, so we have to diff against that and update the dictionary with blank fields appropriately.